cmd/ebitenmobile/_files/EbitenView.java

```diff
@@ -329,11 +363,19 @@ private static int getAxisMask(InputDevice joystickDevice) {
         final int SDL_CONTROLLER_AXIS_TRIGGERRIGHT = 5;
 
         int naxes = 0;
+        boolean haveZ = false;
+        boolean havePastZBeforeRZ = false;
         for (InputDevice.MotionRange range : joystickDevice.getMotionRanges()) {
             if ((range.getSource() & InputDevice.SOURCE_CLASS_JOYSTICK) != 0) {
-                if (range.getAxis() != MotionEvent.AXIS_HAT_X && range.getAxis() != MotionEvent.AXIS_HAT_Y) {
+                int axis = range.getAxis();
+                if (axis != MotionEvent.AXIS_HAT_X && axis != MotionEvent.AXIS_HAT_Y) {
                     naxes++;
                 }
+                if (axis == MotionEvent.AXIS_Z) {
```

Comment:
Note: this code depends on AXIS value ordering (in particular on AXIS_HAT_X and AXIS_HAT_Y not lying between AXIS_Z and AXIS_RZ). SDL's code implicitly has this in the above != block, one indent deeper.

If you want me to move it inside the if() block above that increments naxes, I'll do that - however I find it more readable without likely excessive indent, as I find it unlikely these axis values will ever change.

Reply:
I prefer less indentations so let's keep this as it is.

Reply:
Let's keep it as is rather - that way adding something below won't get confusing.

Of course the two blocks are mutually exclusive anyway, even without `continue` or `else` - I wrote them this way to show a form of logical grouping, that's all.

Reply:
Acknowledged.